src/error.rs

```diff
@@ -59,19 +72,32 @@ pub enum Error {
     /// expected data structure
     ///
     /// [`Config`]: ../engine/config/struct.Config.html
-    DeserializationFailed { source: serde_yaml::Error },
+    DeserializationFailed {
```

Comment:
Per the other comment, given that you've introduced a `JsonDeserializationFailed`, would it make sense to rename this to avoid the appearance of overlap?  Perhaps `YamlDeserializationFailed`?

Reply:
Fixed.